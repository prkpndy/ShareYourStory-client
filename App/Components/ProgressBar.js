import React, {useRef, useState, useEffect} from 'react';
import {View, StyleSheet, Animated} from 'react-native';

function useInterval(callback, delay) {
    const savedCallback = useRef();

    useEffect(() => {
        savedCallback.current = callback;
    }, [callback]);

    useEffect(() => {
        function tick() {
            savedCallback.current();
        }
        if (delay !== null) {
            let id = setInterval(tick, delay);
            return () => clearInterval(id);
        }
    }, [delay]);
}

const ProgressBar = props => {
    let animation = useRef(new Animated.Value(0));
    const [progress, setProgress] = useState(0);
    useInterval(() => {
        if (progress < 100) {
            setProgress(progress + 1);
        } else {
            props.onCompletion();
        }
    }, 50);

    useEffect(() => {
        Animated.timing(animation.current, {
            toValue: progress,
            duration: 100,
            useNativeDriver: false,
        }).start();
    }, [progress]);

    const width = animation.current.interpolate({
        inputRange: [0, 100],
        outputRange: ['0%', '100%'],
        extrapolate: 'clamp',
    });
    return (
        <View style={styles.progressBar}>
            <Animated.View
                style={
                    ([StyleSheet.absoluteFill],
                    {backgroundColor: 'black', width})
                }
            />
        </View>
    );
};

export default ProgressBar;

const styles = StyleSheet.create({
    progressBar: {
        flexDirection: 'row',
        height: 8,
        width: '98%',
        backgroundColor: 'transparent',
        borderColor: '#000',
        borderWidth: 1,
    },
});
