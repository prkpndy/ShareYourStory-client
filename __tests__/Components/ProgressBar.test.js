/**
 * @format
 */

import 'react-native';
import React from 'react';
import ProgressBar from '../../App/Components/ProgressBar';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer
        .create(<ProgressBar onCompletion={() => {}} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
