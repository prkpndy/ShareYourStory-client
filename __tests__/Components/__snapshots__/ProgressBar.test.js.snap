// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "borderColor": "#000",
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 8,
      "width": "98%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "width": "0%",
      }
    }
  />
</View>
`;
